### DISCUSSION TITLE
音频QoS问题

### DISCUSSION THREAD

**Discussion by stopandjoin** [Tencent/tquic#271] [Q&A (问题和回答)] [answered]
@iyangsj 请教一下。 我用tquic传输流媒体，在处理音频QoS时，遇到网络抖动的优化问题，添加音频的jitter buffer时，因为需要根据网络状态动态调整buffer大小。  tquic中有什么状态参数，可以作为一个参考，来调整buffer大小？

**Comment by glime-ai** [comment]
在TQUIC中，您可以利用其提供的丰富的可观测性特性来监控网络状态，并据此调整音频的jitter buffer大小。TQUIC支持灵活的配置和丰富的观测性能，这可以帮助您获取关于网络状况的详细信息，如延迟、丢包率和吞吐量等。

具体来说，您可以关注以下几个方面的参数：
1. **RTT（Round-Trip Time）**: 这是衡量网络延迟的一个重要指标，可以帮助您判断网络的速度和稳定性。
2. **丢包率**: 高丢包率可能会影响音频流的质量，通过监控这一参数，可以适时调整jitter buffer的大小以补偿丢失的数据包。
3. **吞吐量**: 网络的吞吐量变化也是一个重要的参考指标，它可以反映当前网络的拥塞状况。

TQUIC的文档中提到了其支持多种拥塞控制算法和多路径QUIC，这些功能也可以用来优化网络性能和提高传输的稳定性。您可以根据实际应用场景的需要，选择合适的拥塞控制策略。

建议您查阅TQUIC的[API参考文档](https://tquic.net/zh/docs/category/api-reference)和[性能监控相关文档](https://tquic.net/zh/docs/further_readings/benchmark)，以获取更详细的信息和具体的实现方法。

This comment was generated by [Glime](https://glimelab.ai/?utm_source=44520fce49ba43798485a628d6928e22&utm_medium=github).

**Comment by iyangsj** [comment, marked as answer]
tquic没有直接的状态参数用于设置jitter buffer大小，但可以获取网络路径的统计值（例如RTT变化、丢包率、带宽）作为决策依据。

具体取决于你所采用的jitter buffer自适应算法。例如，当检测到显著的RTT波动时，可以增加jitter buffer大小；当网络条件良好时，可以减小Jitter Buffer大小，以尽量减少播放延迟。

**Reply by stopandjoin** [reply]
thanks
